theblues/plans.py

```diff
@@ -54,18 +61,124 @@ def get_plans(self, reference):
                     "%Y-%m-%dT%H:%M:%SZ"
                 ),
                 description=plan.get('description'),
-                price=plan.get('price')), json))
+                price=plan.get('price')), response))
         except (KeyError, TypeError, ValueError) as err:
             log.info(
                 'cannot process plans: invalid JSON response: {!r}'.format(
-                    json))
+                    response))
             raise ServerError(
                 'unable to get list of plans for {}: {}'.format(
                     reference.path(), err))
         except Exception as exc:
             log.info(
                 'cannot process plans: invalid JSON response: {!r}'.format(
-                    json))
+                    response))
             raise ServerError(
                 'unable to get list of plans for {}: {}'.format(
                     reference.path(), exc))
+
+    def list_wallets(self):
+        """Get the list of wallets
+
+        @return an object of wallets, a total, and available credit.
+        @raise ServerError
+        """
+        response = make_request(
+            '{}wallet'.format(self.url),
+            timeout=self.timeout,
+            client=self._client)
+        try:
+            response['wallets'] = tuple(map(
+                lambda wallet: Wallet(
+                    owner=wallet['owner'],
+                    wallet=wallet['wallet'],
+                    limit=wallet['limit'],
+                    budgeted=wallet['budgeted'],
+                    unallocated=wallet['unallocated'],
+                    available=wallet['available'],
+                    consumed=wallet['consumed'],
+                    default='default' in wallet), response['wallets']))
+            total = response['total']
+            response['total'] = WalletTotal(
+                limit=total['limit'],
+                budgeted=total['budgeted'],
+                available=total['available'],
+                unallocated=total['unallocated'],
+                usage=total['usage'],
+                consumed=total['consumed'])
+            return response
+        except (KeyError, TypeError, ValueError, Exception) as err:
```

Comment:
Exception is the base for all exceptions, so if we are catching that it's not sueful to specify the others. Maybe separate them in two except blocks?